pkg/providers/tinkerbell/assert_test.go

```diff
@@ -229,6 +229,41 @@ func TestAssertTinkerbellIPAndControlPlaneIPNotSame_SameFails(t *testing.T) {
 	g.Expect(tinkerbell.AssertTinkerbellIPAndControlPlaneIPNotSame(clusterSpec)).ToNot(gomega.Succeed())
 }
 
+func TestAssertPortsNotInUse_Succeeds(t *testing.T) {
+	g := gomega.NewWithT(t)
+	ctrl := gomock.NewController(t)
+
+	netClient := mocks.NewMockNetClient(ctrl)
+	netClient.EXPECT().
+		DialTimeout(gomock.Any(), gomock.Any(), gomock.Any()).
+		AnyTimes().
+		Return(nil, errors.New("failed to connect"))
```

Comment:
actually, this will be called 3 times exactly so you can do something like this instead
```suggestion
		Return(nil, errors.New("failed to connect")).Times(3)
```